src/controllers/connectionManager.ts

```diff
@@ -289,6 +289,18 @@ export default class ConnectionManager {
 		return ConnectionCredentials.createConnectionDetails(connectionInfo);
 	}
 
+	/**
+	 * Send a request to the SQL Tools Server client
+	 * @param type The of the request to make
+	 * @param params The params to pass with the request
+	 * @returns A thenable object for when the request receives a response
+	 */
+	public sendRequest<P, R, E, R0>(requestType: RequestType<P, R, E, R0>, params?: P): Thenable<R> {
```

Comment:
We generally shouldn't be using Thenables in our code - that's only something we do for extension APIs. This should be returning `Promise<R>`

Reply:
The issue here is that the returned sendRequest of type Thenable ([serverclient.ts](https://github.com/microsoft/vscode-mssql/blob/main/src/languageservice/serviceclient.ts#L380)) which then also calls the sendRequest from vscode-languageclient. They are all of type Thenable.

Reply:
You can solve this by making this function async, awaiting the call to sendRequest and then returning the result